### PR TITLE
[WIP]  Alternative take on boosted queries

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

**Comment by audunhalland** [comment]
Currently `BoostQuery` isn't `pub use`, so I can't test this branch in my own project yet

**Comment by fulmicoton** [comment]
@audunhalland Now it is pub.

File: src/query/query.rs [hunk 1]
```diff
@@ -104,4 +105,35 @@ impl QueryClone for Box<dyn Query> {
     }
 }
 
+pub struct BoostQuery {
```
Comment (audunhalland) [inline src/query/query.rs]: Probably this needs a `::new` function

**Comment by fulmicoton** [comment]
Added. `QueryParser` support too.

**Pull request merged**